torch/_inductor/scheduler.py

```diff
@@ -3050,11 +3062,17 @@ def _codegen(self) -> None:
                 if device is not None and self.get_backend(device).ready_to_flush():
                     self.flush()
 
+            if config.annotate_buffers:
+                V.graph.wrapper_code.writeline("nvtx.device_range_end(buffer_annotation)")
+
         if self.current_device and device_need_guard(self.current_device.type):
             # exit the outermost CUDA device guard. this is
             # important for nested indentation codegen-ing.
             V.graph.wrapper_code.codegen_device_guard_exit()
 
+        if config.annotate_training:
+            V.graph.wrapper_code.writeline("nvtx.device_range_end(training_annotation)")
```

Comment:
would you mind posting an example output code for a fused kernel ? this would also be a good candidate for a test see `run_and_get_code`

Reply:
Added a couple of test cases here https://github.com/pytorch/pytorch/pull/130429/commits/7d623efa64a4834714eca3696802b5f9bf018452
Example output is in the comment below: https://github.com/pytorch/pytorch/pull/130429#issuecomment-2291641981